crates/viewer/re_viewer_context/src/drag_and_drop.rs

```diff
@@ -0,0 +1,184 @@
+//! Implement a global drag-and-drop payload type that enable dragging from various parts of the UI
+//! (e.g., from the streams tree to the viewport, etc.).
+
+use std::fmt::Formatter;
+
+use itertools::Itertools;
+
+use re_ui::{
+    ColorToken, Hue,
+    Scale::{S325, S375},
+    UiExt,
+};
+
+use crate::{Contents, Item, ItemCollection};
+
+//TODO(ab): add more type of things we can drag, in particular entity paths
+#[derive(Debug)]
+pub enum DragAndDropPayload {
+    /// The dragged content is made only of [`Contents`].
+    Contents { contents: Vec<Contents> },
+
+    /// The dragged content is made of a collection of [`Item`]s we do know how to handle.
+    Invalid,
+}
+
+impl DragAndDropPayload {
+    pub fn from_items(selected_items: &ItemCollection) -> Self {
+        if let Ok(contents) = selected_items.try_into() {
+            Self::Contents { contents }
+        } else {
+            Self::Invalid
+        }
+    }
+}
+
+impl std::fmt::Display for DragAndDropPayload {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::Contents { contents } => items_to_string(
+                contents
+                    .iter()
+                    .map(|content| content.as_item())
+                    .collect_vec()
+                    .iter(),
+            )
+            .fmt(f),
+
+            // this is not used in the UI
+            Self::Invalid => "invalid selection".fmt(f),
+        }
+    }
+}
+
+/// Display the currently dragged payload as a pill in the UI.
+///
+/// This should be called once per frame.
+pub fn drag_and_drop_payload_cursor_ui(ui: &mut egui::Ui) {
+    if let Some(payload) = egui::DragAndDrop::payload::<DragAndDropPayload>(ui.ctx()) {
+        if let Some(pointer_pos) = ui.ctx().pointer_interact_pos() {
```

Comment:
It's unclear what `ui` should be passed in here, and indeed: it doesn't seem to matter.

I think it would be more clear to pass in a `egui::Context` and then create a new `Ui` with `Ui::new`